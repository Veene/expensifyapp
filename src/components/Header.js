import React, {Component} from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = (props) => (
    <header className="header">
        <div className="content_container">
            <div className="header__content">
                <Link className="header__title" to='/dashboard' activeClassName="is-active">
                <h1>Expensify</h1>
                </Link>
                <button onClick={props.startLogout}>Logout</button>
            </div>
            
        </div>
        
    </header>
)

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
