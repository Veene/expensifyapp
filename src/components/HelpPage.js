import React, {Component} from 'react';

const HelpPage = () => (
    <div>
        This is from my HelpPage component
    </div>
)

export default HelpPage;