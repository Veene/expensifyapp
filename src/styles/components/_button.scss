.button {
    background: $blue;
    border: none;
    color: white;
    padding: $sm-size;
    font-weight: 300;
    font-size: $font-size-large;
}