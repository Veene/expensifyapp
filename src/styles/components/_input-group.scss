.input-group {
    display: flex;
    flex-direction: column;
    margin-bottom: $lg-size;

    @media (min-width: $desktop-breakpoint) {
        flex-direction: row;
        margin-bottom: $lg-size;
    }
}
.input-group__item {
    margin-right: $sm-size;
    margin-bottom: $sm-size;

    @media (min-width: $desktop-breakpoint) {
        margin: 0 $sm-size 0 0;
    }
}