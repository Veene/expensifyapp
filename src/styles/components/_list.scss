.list-header {
    background: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $sm-size $med-size;
}
.list-body {
    margin-bottom: $med-size;
    @media (min-width: $desktop-breakpoint) {
        margin-bottom: $lg-size;
    }
}
.list-item {
    display: flex;
    flex-direction: column;
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    padding: $sm-size;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }
    @media (min-width: $desktop-breakpoint) {
        align-items: center;
        flex-direction: row; 
        justify-content: space-between; 
        padding: $med-size;
    }
}
.list-item--message {
    align-items: center;
    justify-content: center;
    color: $grey;
    padding: $med-size;
    &:hover {
        background: none;
    }
}
.list-item__title {
    margin: 0;
    word-break: break-all;
}
.list-item__sub-title {
    color: $grey;
    font-size: $font-size-small;
}
.list-item__data {
    margin: $sm-size 0 0 0;
    @media (min-width: $desktop-breakpoint) {
        margin: 0;
        padding-left: $sm-size;
    }
}