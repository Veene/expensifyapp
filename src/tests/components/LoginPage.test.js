import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should render snapshot of LoginPage', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
})