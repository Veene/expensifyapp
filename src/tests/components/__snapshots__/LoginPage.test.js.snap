// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render snapshot of LoginPage 1`] = `
<div>
  <button>
    Login
  </button>
</div>
`;
